### DISCUSSION TITLE
Fix FindPkgConfig.cmake for multi config generators.

### DISCUSSION THREAD

**Discussion by Neumann-A** [microsoft/vcpkg#17931] [Ideas]
Context: https://cmake.org/cmake/help/latest/module/FindPkgConfig.html

Problem:
The FindPkgConfig module is not configuration aware. As such it will not work correctly with multi config generators.

Solution:
simply adjust the module to do two configuration aware calls and setup targets/variables accordingly.

**Comment by JackBoosY** [comment]
@Neumann-A I'm not familiar with pkgconfig, can you open a PR to solve this?

Thanks.

**Comment by Neumann-A** [comment]
that probably needs to be added to `/scripts/buildsystems/cmake/FindPkgConfig.cmake` and the path  `/scripts/buildsystems/cmake/ `added to the module path within the toolchain. I don't see compilin argument to try and make a port out of it.
